sympy/matrices/matrices.py

```diff
@@ -3966,6 +3966,72 @@ def vech(self, diagonal=True, check_symmetry=True):
                     count += 1
         return v
 
+    def upper(a, diag=0):
```

Comment:
`diag` should be a keyword only argument, i.e., change to `upper(a, *, diag=0)`. I also think `diag_offset` would be a better name.